libvast/src/taxonomies.cpp

```diff
@@ -26,42 +26,49 @@
 
 namespace vast {
 
-static caf::error to_concepts_impl(const data& d, concepts_type& out) {
-  if (const auto& tl = caf::get_if<list>(&d)) {
-    for (const auto& item : *tl) {
+caf::error convert(const data& d, concepts_type& out) {
+  if (const auto& c = caf::get_if<record>(&d)) {
```

Comment:
If `d` is a `const data` then `get_if<record>` will return a `const record&`, that can't be assigned to `auto&`. If we use just `auto` it will create a copy, so the only way to get rid of the const is to use `decltype(auto) c = caf::get_if<record>(&d)`, and I don't like that.

Reply:
Fair enough.